activation/atxdb.go

```diff
@@ -192,19 +244,22 @@ func epochCounterKey(ech types.EpochId) []byte {
 func (db *ActivationDb) incValidAtxCounter(ech types.EpochId) error {
 	key := epochCounterKey(ech)
 	val, err := db.atxs.Get(key)
-	if err == nil {
-		return db.atxs.Put(key, common.Uint64ToBytes(common.BytesToUint64(val)+1))
+	if err != nil {
+		return db.atxs.Put(key, common.Uint32ToBytes(1))
 	}
-	return db.atxs.Put(key, common.Uint64ToBytes(1))
+	return db.atxs.Put(key, common.Uint32ToBytes(common.BytesToUint32(val)+1))
 }
 
-func (db *ActivationDb) ActiveIds(ech types.EpochId) uint64 {
+func (db *ActivationDb) ActiveSetSize(ech types.EpochId) uint32 {
 	key := epochCounterKey(ech)
+	db.RLock()
 	val, err := db.atxs.Get(key)
+	db.RUnlock()
 	if err != nil {
+		//0 is not a valid active set size
 		return 0
 	}
-	return common.BytesToUint64(val)
+	return common.BytesToUint32(val)
 }
 
 func (db *ActivationDb) addAtxToEpoch(epochId types.EpochId, atx types.AtxId) error {
```

Comment:
i'd consider adding comments that addAtxToEpoch and addAtxToNodeId should only be used inside a method that invokes db.Lock(), the way it's implemented now its not that obvious